### PR TITLE
fix(bundler): More flexible parsing for gem lines

### LINKED ISSUE #7318: Gemfile source block not parsed
**What Renovate type, platform and version are you using?**

CLI

**Describe the bug**

The following source block is not parsed in `Gemfile`:

```
source 'https://hub.tech.my.domain.de/artifactory/api/gems/my-gems-prod-local/' do
    gem 'sfn_my_dep1', "~> 1"
    gem 'sfn_my_dep2', "~> 1"

end
```

### PR DESCRIPTION
**Pull request by zharinov**
Closes #7318

### CODE REVIEW

File: lib/manager/bundler/__snapshots__/extract.spec.ts.snap [hunk 1]
```diff
@@ -2491,12 +2491,13 @@ Object {
       },
     },
     Object {
+      "currentValue": "~> 0.0.2",
+      "datasource": "rubygems",
```
Comment (zharinov) [inline lib/manager/bundler/__snapshots__/extract.spec.ts.snap:2495]: https://github.com/renovatebot/renovate/blob/master/lib/manager/bundler/__fixtures__/Gemfile.gitlab-foss#L47

File: lib/manager/bundler/extract.ts [hunk 2]
```diff
@@ -38,33 +38,18 @@ export async function extractPackageFile(
...
-          currentValue.split(gemDelimiter).length === 3
-        ) {
-          currentValue = currentValue.slice(1, -1);
-        }
-        dep.currentValue = currentValue;
+      if (gemMatch.groups.currentValue) {
+        const currentValue = gemMatch.groups.currentValue;
+        dep.currentValue = /\s*,\s*/.test(currentValue)
+          ? currentValue
+          : currentValue.slice(1, -1);
```
Comment (zharinov) [inline lib/manager/bundler/extract.ts:52]: Preserve current behavior of range extraction, if needed: 
`gem "listen", ">= 3.0.5", "< 3.2", require: false` -> `currentValue === '">= 3.0.5", "< 3.2"'`

File: lib/manager/bundler/extract.ts [hunk 3]
```diff
@@ -38,33 +38,18 @@ export async function extractPackageFile(
...
-    let gemMatch: RegExpMatchArray;
-    let gemDelimiter: string;
-    for (const delimiter of delimiters) {
-      const gemMatchRegex = `^gem ${delimiter}([^${delimiter}]+)${delimiter}(,\\s+${delimiter}([^${delimiter}]+)${delimiter}){0,2}`;
-      if (regEx(gemMatchRegex).test(line)) {
-        gemDelimiter = delimiter;
-        gemMatch = gemMatch || regEx(gemMatchRegex).exec(line);
-      }
-    }
+    const gemMatchRegex = /^\s*gem\s+(['"])(?<depName>[^'"]+)\1(\s*,\s*(?<currentValue>(['"])[^'"]+\5(\s*,\s*\5[^'"]+\5)?))?/;
```
Comment (rarkins) [inline lib/manager/bundler/extract.ts:41]: This will now allow unbalanced quotes, but would that even happen except for in an incorrect Gemfile?

Comment (rarkins) [reply lib/manager/bundler/extract.ts:41]: I see the gitlab example you called out. I guess that tradeoff is worth it.

**Review by rarkins — CHANGES_REQUESTED** [review]
Can you add a new fixture like the one in the issue?

File: lib/manager/bundler/extract.ts (hunk 3 above)
Comment (zharinov) [reply lib/manager/bundler/extract.ts:41]: I would extract unbalanced/heterogeneous quotes, but not sure if versioning scheme can deal with it

**Review by rarkins — APPROVED** [review]

**Pull request merged**